src/robotFactory.js

```diff
@@ -1,10 +1,85 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: 0,
+      y: 0,
+      ...coords,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const zAxis = {
+      z: 0,
+      ...coords,
+    };
+
+    super(name, weight, zAxis, chipVersion);
+  }
```

Comment:
зачем базовому роботу 3я координата? она же нужна только летающим
```suggestion
    super(name, weight, coords, chipVersion);
    this.coords = {
      z: 0,
      ...coords,
    };
  }
```